iaso/tests/gpkg/test_import_with_sub_org_unit_type.py

```diff
@@ -0,0 +1,31 @@
+from iaso.gpkg.import_gpkg import import_gpkg_file
+from iaso.models import Account, Project
+from iaso.test import APITestCase
+from setuper.pyramid import update_org_unit_sub_type
+
+
+class OrgUnitImportFromGPKG(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.account = Account.objects.create(name="a")
+        cls.user_test = cls.create_user_with_profile(username="test", account=cls.account)
+        cls.project = Project.objects.create(name="Project 1", account=cls.account, app_id="test_app_id")
+
+    def test_minimal_import_with_sub_org_unit_type(self):
+        import_gpkg_file(
+            "./iaso/tests/fixtures/gpkg/minimal.gpkg",
+            project_id=self.project.id,
+            source_name="test",
+            version_number=1,
+            validation_status="new",
+            description="",
+        )
+        self.client.force_authenticate(self.user_test)
+        response = self.client.get("/api/v2/orgunittypes/")
+        self.assertJSONResponse(response, 200)
+        response_data = response.json()
+
+        updated_with_sub_types = update_org_unit_sub_type(self.client, self.project.id, response_data["orgUnitTypes"])
+
+        for org_unit_type_with in updated_with_sub_types:
+            self.assertJSONResponse(org_unit_type_with, 200)
```

Comment:
Do we know the expected new sub types here?
If so we should test that too in addition to testing the returned HTTP code.

Reply:
I am wondering for testing the expected new sub type, if from the [line 27](https://github.com/BLSQ/iaso/pull/1346/files/71e05adda38288607769ccd930683179e4f3ff1e#diff-b8927c3582b607cd48225f0e40d6a3efd40efd9112d7280a75ab75fc8afae0ebR27)  we filter on the expected new sub types with ;

```
expected_new_sub_types = []
for org_unit_type in response_data["orgUnitTypes"]:
    org_unit_type_level = org_unit_type["depth"]
    sub_unit_type_ids = [
        org_unit_type["id"]
        for org_unit_type in response_data["orgUnitTypes"]
        if org_unit_type["depth"] == (org_unit_type_level + 1)
    ]
    expected_new_sub_types.append(sub_unit_type_ids)
```
But, it looks a re-implement of the logic of **update_org_unit_sub_type**

Then, from  [line 29](https://github.com/BLSQ/iaso/pull/1346/files/71e05adda38288607769ccd930683179e4f3ff1e#diff-b8927c3582b607cd48225f0e40d6a3efd40efd9112d7280a75ab75fc8afae0ebR29)  

```
updated_sub_types = []
for org_unit_type_with in updated_with_sub_types:
    self.assertJSONResponse(org_unit_type_with["updated_org_unit_typs"], 200)
    updated_sub_types.append(org_unit_type_with["sub_unit_type_ids"][0])

sub_types_ids = reduce(lambda a, b: a + b, expected_new_sub_types)
self.assertEqual(updated_sub_types, sub_types_ids)
```

Reply:
TBH I don't know if it's doable?

My point is that In general, we know what we are expecting from a test:

```python
expected_result = "foo"

result = test_my_function()

assert result == expected_result
```

Here, the only thing that you are testing is that the API call returns 200, but we don't know if the data was modified in the expected way.

We can discuss this tomorrow, it may be easier to talk about it.

Reply:
Are we really doing tests on the setuper? I'm thinking this might be overkill.